### ISSUE TITLE
Oplock Documentation Disjointed and do not reference one another

### ISSUE THREAD

**Issue by ECartman** [MicrosoftDocs/feedback#3924]
there are 2 articles that lack information. 
[Using oplocks for network redirector performance](https://learn.microsoft.com/en-us/windows-hardware/drivers/ifs/oplock-overview#types-of-oplocks) and [Types of Opportunistic Locks](https://learn.microsoft.com/en-us/windows/win32/fileio/types-of-opportunistic-locks)

one of the articles Lack the information from the other. and at the end they don't provide enough insight. 
what is more troublesome is that. depending on what are you researching or which keywords you use you will reach one side and not the other. 
therefore, information from one Article should be pass to the other. OR better yet merge, update and centralize the articles for Opportunistic locks.     

Information From 
[FileIO](https://github.com/MicrosoftDocs/win32/tree/docs/desktop-src/FileIO) 
is missing: 
the historic details such as: 

> The following four oplocks were implemented in Windows NT 3.1 (Level 1, Level 2, Batch) and Windows 2000 (Filter), and are considered "legacy oplocks"

or the mention of: 

> The following oplocks were added in Windows 7, and are collectively known as "Windows 7 oplocks":
> 
> A Read (R) oplock (shared) indicates that there are multiple readers of a stream and no writers. This supports client read caching.
> 
> A Read-Handle (RH) oplock (shared) indicates that there are multiple readers of a stream, no writers, and that a client can keep a stream open on the server even though the local accessor on the client machine has closed the stream. This supports client read caching and handle caching.
> 
> A Read-Write (RW) oplock (exclusive) allows a client to open a stream for exclusive access and allows the client to perform arbitrary buffering. This supports client read caching and write caching.
> 
> A Read-Write-Handle (RWH) oplock (exclusive) allows a client to keep a stream open on the server even though the local accessor on the client machine has closed the stream. This supports client read caching, write caching, and handle caching.

[IFS](https://github.com/MicrosoftDocs/windows-driver-docs/blob/staging/windows-driver-docs-pr/ifs)
miss most of the rest of the information on how to break them. etc. 

**Describe the solution you'd like**
Merge the articles.([Using oplocks for network redirector performance](https://learn.microsoft.com/en-us/windows-hardware/drivers/ifs/oplock-overview#types-of-oplocks) and [Types of Opportunistic Locks](https://learn.microsoft.com/en-us/windows/win32/fileio/types-of-opportunistic-locks))_ ensure the information is accurate and sufficient. and link the topics where needed. 

**Describe alternatives you've considered**
document a list of documents that MSFT should centralize... 

**Additional context**

[win32](https://github.com/MicrosoftDocs/win32/tree/docs)/[desktop-src](https://github.com/MicrosoftDocs/win32/tree/docs/desktop-src)/[FileIO](https://github.com/MicrosoftDocs/win32/tree/docs/desktop-src/FileIO)
/[types-of-opportunistic-locks.md](https://github.com/MicrosoftDocs/win32/blob/docs/desktop-src/FileIO/types-of-opportunistic-locks.md)

**Comment by gewarren** [comment]
@lorihollasch FYI

**Comment by lorihollasch** [comment]
Thanks for the issue, @ECartman. These two repos are for different audiences (SDK is for app developers; WDK is for driver developers), so the articles are for the most part targeted as such. In this particular instance I agree that there is some overlap, so adding cross-references between the WDK and SDK is useful. I've added such a link to the WDK's Oplock Overview page. I also edited the WDK's oplock articles for better clarity and added links to make it easier to discover info about granting, breaking, etc. The SDK team has opened an internal work item to update their oplock pages (thanks, @alvinashcraft).
